Add ajax jquery to upvote downvote

trishzag opened the pull request:
Added ajax/jquery to upvote and downvote buttons.
Amended show page and tests to reflect the code changes.
Added selenium-webdriver to allow capybara to test the js actions.  Note that this requires firefox installed in applications folder in order for test to function correctly.

trishzag commented on app/assets/javascripts/voting.js:
```diff
@@ -0,0 +1,29 @@
+  })
+  .done(function(data){
+    $($t.siblings("#score")).text(data);
+  })
```
the code does not work correctly if I add in semicolons as hound suggests.

trishzag commented on app/assets/stylesheets/application.css.scss:
```diff
@@ -0,0 +1,50 @@
+  background-size: cover;
+}
+
+#user_profile {
```
As this is not my code, and was previously merged as it is, I am uncomfortable changing this.

trishzag commented on app/assets/stylesheets/application.css.scss:
```diff
@@ -0,0 +1,50 @@
+}
+
+.side-bar {
+  background-color: #333333;
```
Same note as above.

trishzag commented on app/assets/javascripts/voting.js:
```diff
@@ -0,0 +1,29 @@
+  })
+  .done(function(data){
+    $($t.siblings("#score")).text(data);
+  })
```
Hound hates the lack of semicolons but code doesn't work if I add them in.

trishzag commented on app/assets/javascripts/voting.js:
```diff
@@ -0,0 +1,29 @@
+  })
+  .done(function(data){
+    $($t.siblings("#score")).text(data);
+  })
```
^^ see comment above

trishzag commented:
Note that the hound violations are requests for semicolons that will break code and two requests that are for code that had already been merged (css changes made on a different branch).

radavis commented on spec/rails_helper.rb:
```diff
@@ -66,3 +67,14 @@
+  def self.connection
+    @@shared_connection || retrieve_connection
+  end
+end
```
What is going on, here?

radavis commented on spec/spec_helper.rb:
```diff
@@ -1,5 +1,6 @@
 require 'coveralls'
 require 'support/login_helper'
+require 'selenium-webdriver'
```
If you are requiring this in the `rails_helper.rb` you don't need it here.

radavis commented on spec/rails_helper.rb:
```diff
@@ -66,3 +67,14 @@
 end
+
+class ActiveRecord::Base
+  mattr_accessor :shared_connection
```
`mattr_accessor` isn't a thing.

radavis commented on spec/features/user_downvotes_a_review_spec.rb:
```diff
@@ -1,4 +1,5 @@
 require 'rails_helper'
+require 'selenium-webdriver'
```
since this is in the rails_helper, you don't need it here.

radavis commented on spec/features/user_upvotes_a_review_spec.rb:
```diff
@@ -1,4 +1,5 @@
 require 'rails_helper'
+require 'selenium-webdriver'
```
delete this line

trishzag commented:
Deleted extraneous require "selenium-webdriver" per Richard's comments. 
Also moved two lines of code up to reduce hound comments (for missing semicolon); verified that tests pass and code is working on localhost.
